iDiff/utils/fs_utils.go

```diff
@@ -6,10 +6,22 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 
 	"github.com/golang/glog"
 )
 
+func GetDirectorySize(path string) (int64, error) {
+	var size int64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			size += info.Size()
+		}
```

Comment:
the way filepath.Walk() works is inherently recursive because it crawls down every possible path in the root path passed to it, so you don't have to do any explicit recursion